### ISSUE TITLE
SPACK_SHELL not being set to actual shell when using spack bootstrap

### ISSUE THREAD

**Issue by jrood-nrel** [spack/spack#7441]
### Expected Result

`SPACK_SHELL=bash`, etc

### Actual Result

`SPACK_SHELL=<name of script in which it is running>`

### Steps to reproduce the issue

Run `setup-env.sh` in a script after using `spack bootstrap`.

```
[user@machine ~]$ cat script.sh 
#!/bin/bash
export SPACK_ROOT=${HOME}/spack
source ${SPACK_ROOT}/share/spack/setup-env.sh
echo $SPACK_SHELL
[user@machine ~]$ ./script.sh 
script.sh
```

### Information on your system

Centos 7. Can't seem to recreate this problem on OSX.

### Other Notes

The code which sets `SPACK_SHELL` is as such:
```
function _spack_determine_shell() {
        ps -p $$ | tail -n 1 | awk '{print $4}' | sed 's/^-//' | xargs basename
}
export SPACK_SHELL=$(_spack_determine_shell)
```

Maybe there is good reason to do it that way instead of extracting it out of `$SHELL`, but I think it might need a more robust function to resolve the current shell so it works within a script.

Edit: `export SPACK_SHELL=$(basename $SHELL)` maybe?

**Comment by adamjstewart** [comment]
I'm having trouble reproducing your original problem. With the following file:
```
$ cat script.sh 
. setup-env.sh
echo $SPACK_SHELL
```
I get:
```
$ ./script.sh 
bash
```

**Comment by jrood-nrel** [comment]
@adamjstewart Thanks for looking at this. So it looks like I didn't test OSX when I made this issue. I am not able to recreate it in OSX. It looks like the machine I was on is Centos 7, but maybe there is something else about the machine configuration causing it. Here is me reproducing it on the Centos 7 machine:

```
[user@machine ~]$ cat script.sh 
#!/bin/bash
export SPACK_ROOT=${HOME}/spack
source ${SPACK_ROOT}/share/spack/setup-env.sh
echo $SPACK_SHELL
[user@machine ~]$ ./script.sh 
script.sh
```

**Comment by tgamblin** [comment]
I believe this (the detection, at least) was fixed in #11994.  The detection @carns introduced in #9712 here was not quite right.  Specifically, [this line](https://github.com/spack/spack/pull/9712/files#diff-3f450f9072969bd3b01bdcf9d5142ed9R245) reports `sh` if `bash` is run through a symlink/hard link/filename called `sh`.

**Comment by tgamblin** [comment]
Note that I removed `SPACK_SHELL`, as I do not think we should be exporting variables that tell people which shell they're running.  They won't be correct if you, e.g., launch a subshell, and scripts should do that detection themselves.
